File: Daily/March/goMarch/LeetCode504.go
package goMarch

func ConvertToBase7(num int) string {
	if num == 0 {
		return "0"
	}

	isNegative := num < 0
	if isNegative {
		num = -num
	}

	res := []byte{}

	for num > 0 {
		res = append(res, '0'+byte(num%7))
		num /= 7
	}
	if isNegative {
		res = append(res, '-')
	}

	for i, n := 0, len(res); i < n/2; i++ {
		res[i], res[n-1-i] = res[n-1-i], res[i]
	}
	return string(res)
}

File: Daily/March/LeetCode504.cc
#include<iostream>
#include<algorithm>
#include<string>
using namespace std;

class Solution {
public:
    string convertToBase7(int num) {
        if(num == 0){
            return "0";
        }
        bool isNegative = num < 0; // 负数
        num = abs(num); // 取绝对值
        string res;
        while(num > 0){
            res.push_back(num%7+'0');
            num /= 7;
        }
        if(isNegative){
            res.push_back('-');
        }
        reverse(res.begin(),res.end());
        return res;
    }
};